pandas/tests/test_strings.py

```diff
@@ -938,20 +938,13 @@ def test_match(self):
         exp = Series([True, NA, False])
         tm.assert_series_equal(result, exp)
 
-        # test passing as_indexer still works but is ignored
+        # GH 22316 test the removal of as_indexer from match
         values = Series(['fooBAD__barBAD', NA, 'foo'])
         exp = Series([True, NA, False])
-        with tm.assert_produces_warning(FutureWarning):
+        with tm.assert_raises_regex(TypeError,
+                                    "match() got an unexpected "
+                                    "keyword argument 'as_indexer'"):
```

Comment:
I like the thought.  However, we generally just drop all of the relevant tests when a keyword is getting dropped.